argopy/xarray.py

```diff
@@ -547,3 +583,68 @@ def profile2point(self):
         ds.argo._add_history('Transformed with profile2point')
         ds.argo._type = 'point'
         return ds
+
+    def interp_std_levels(self, std_lev):
+        """ Returns a new dataset interpolated to new inputs levels """
+
+        if self._type != 'profile':
+            raise InvalidDatasetStructure(
+                "Method only available to a collection of profiles")
+
+        # if(self._mode != 'standard'):
+        #    raise InvalidDatasetStructure(
+        #        "Method only available for the standard mode yet")
+
+        if (type(std_lev) is np.ndarray) | (type(std_lev) is list):
+            std_lev = np.array(std_lev)
+            if (np.any(sorted(std_lev) != std_lev)) | (np.any(std_lev < 0)):
+                raise ValueError(
+                    'Standard levels must a numpy array of positive and sorted values')
+        else:
+            raise ValueError('Standard levels must a numpy array or a list')
+
+        ds = self._obj
+
+        # Filtering on pressure levels to avoid extrapolation, this is open to discussion...
+        # i1 = (ds['PRES'].min('N_LEVELS') <= std_lev[0]) & (
+        #    ds['PRES'].max('N_LEVELS') >= std_lev[-1])
+        i1 = (ds['PRES'].max('N_LEVELS') >= std_lev[-1])
+        dsp = ds.where(i1, drop=True)
+
+        # check if any profile is left
+        if (len(dsp['N_PROF']) == 0):
+            raise ValueError(
+                'Zero profiles left. Some levels may be to deep for selected profiles')
+
+        # add new vertical dimensions, this has to be in the datasets to apply ufunc later
+        dsp['Z_LEVELS'] = xr.DataArray(std_lev, dims={'Z_LEVELS': std_lev})
+
+        # init
+        ds_out = xr.Dataset()
+
+        # vars to interpolate
+        datavars = [dv for dv in list(dsp.variables) if set(['N_LEVELS', 'N_PROF']) == set(
+            dsp[dv].dims) and 'QC' not in dv and 'ADJUSTED' not in dv]
```

Comment:
**Standard vs expert modes**

So what about the ADJUSTED and ADJUSTED_ERROR fields ?
The function allows an expert dataset as input, but it returns fields as for an standard user, this may be confusing.

Is it also the case that it will use PRES whatever the user mode as a vertical axis to interpolate ?
Interpolation in expert mode is rather complex (should we use PRES or PRES_ADJUSTED ?), may be this new function should be limited to standard users at this point ?

Or may be it could take as optional input the name of the vertical axis to use ?

Reply:
I'm still facing some issues with the interpolation in expert mode, some difficulties with xarray/dask and the method applied to ADJUSTED parameters. So for now yes I think standard mode only is a good idea.